File: src/TestApp_MassTransit/OrderSubmittedConsumer.cs
using MassTransit;
using Microsoft.Extensions.Logging;

namespace TestApp;

class OrderSubmittedConsumer :
    IConsumer<OrderSubmitted>
{
    private readonly ILogger<OrderSubmittedConsumer> _logger;

    public OrderSubmittedConsumer(ILogger<OrderSubmittedConsumer> logger)
    {
        _logger = logger;
    }

    public Task Consume(ConsumeContext<OrderSubmitted> context)
    {
        var message = context.Message;

        _logger.LogInformation("📨 Order submitted: {OrderId} ✅", message.OrderId);

        return Task.CompletedTask;
    }
}

File: src/Java/testapp/src/main/java/com/myservicebus/testapp/OrderSubmittedConsumer.java
package com.myservicebus.testapp;

import java.util.concurrent.CompletableFuture;

import com.google.inject.Inject;
import com.myservicebus.ConsumeContext;
import com.myservicebus.Consumer;
import com.myservicebus.logging.Logger;
import com.myservicebus.logging.LoggerFactory;

class OrderSubmittedConsumer implements Consumer<OrderSubmitted> {
    private final Logger logger;

    @Inject
    public OrderSubmittedConsumer(LoggerFactory loggerFactory) {
        this.logger = loggerFactory.create(OrderSubmittedConsumer.class);
    }

    @Override
    public CompletableFuture<Void> consume(ConsumeContext<OrderSubmitted> context) throws Exception {
        var message = context.getMessage();
        var orderId = message.getOrderId();
        var replica = message.getReplica();

        logger.info("📨 Order submitted: {} by {} ✅", orderId, replica);

        return CompletableFuture.completedFuture(null);
    }
}
